### PR TITLE
build: Upgrade all dependencies

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description
Updates runtime and testing dependencies to their latest versions based on the channels we were following.

## :green_heart: How did you test it?
Ran instrumentation and unit tests, did a functional test manually.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code

### CODE REVIEW

File: app/src/main/java/com/zeapo/pwdstore/Application.kt [hunk 1]
```diff
@@ -25,7 +24,6 @@ class Application : android.app.Application(), SharedPreferences.OnSharedPrefere
         prefs = PreferenceManager.getDefaultSharedPreferences(this)
         if (BuildConfig.ENABLE_DEBUG_FEATURES || prefs?.getBoolean("enable_debug_logging", false) == true) {
             plant(DebugTree())
-            WhatTheStack(this).init()
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/Application.kt:28]: Is this activated automatically so we no longer need this line?

Comment (FabianHenneke) [reply app/src/main/java/com/zeapo/pwdstore/Application.kt:28]: My bad, saw the commit message now.

**Comment by msfjarvis** [comment]
> While you are at it: When I ran Android Lint earlier, I noticed that we have plenty of unused resources that we could remove.

I'll get that done.

**Comment by msfjarvis** [comment]
> While you are at it: When I ran Android Lint earlier, I noticed that we have plenty of unused resources that we could remove.

Lint appears to be blisfully unaware of ViewBinding and goes trigger happy, removing all layouts that use ViewBinding as well as all `@+id/` declarations. Maybe not worth trusting at the moment...

**Comment by msfjarvis** [comment]
@FabianHenneke have you tested any builds with this? I wanna get it in well before the release deadline so we'll be able to crash edge case regressions from snapshot users early. FWIW I haven't faced any bugs, but there's a whole slew of deprecations from `start<foo>FromResult`.

**Comment by FabianHenneke** [comment]
> @FabianHenneke have you tested any builds with this? I wanna get it in well before the release deadline so we'll be able to crash edge case regressions from snapshot users early. FWIW I haven't faced any bugs, but there's a whole slew of deprecations from `start<foo>FromResult`.

The deprecations around `startActivityFromResult` should not affect us for quite some time. The new API for that is rather complicated and does not seem to be well set-up for our use cases. It's probably safe to ignore for at least a few Android versions, but might be worth looking into when switching the OpenKeychain auth flow over to SSHJ.

I have build successfully with this PR and will look for bugs tomorrow. I may have found one, but will have to investigate it more.

**Comment by FabianHenneke** [comment]
I went through all my daily workflows and did not find any bugs. 

Yesterday, when I used specifically the Autofill generate action, the Password Store `UserPreference` activity would open up instead of `PgpActivity`. I have not been able to reproduce this on a clean rebuild on two phones though. It was probably just a build issue caused by switching configs without a full rebuild.

**Pull request merged**